### PR TITLE
Implement Lambda Event Source Filter update in CFn Provider

### PR DESCRIPTION
**Pull request by Morijarti**
## Motivation
At the moment, lambda event filtering does not support updates, which is causing issues for several costumers using SAM.

## Changes
- Added snapshot test for filter updating
- Added Update support for Lambda EventSourceMapping

## Testing
- New tests have been added to cfn lambda provider

### CODE REVIEW

File: tests/aws/services/cloudformation/resources/test_lambda.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def test_lambda_w_dynamodb_event_filter(deploy_cfn_template, aws_client):
             "FunctionName": function_name,
             "TableName": table_name,
             "Filter": '{"eventName": ["MODIFY"]}',
-        },
+        }
```
Comment (joe4dev) [inline tests/aws/services/cloudformation/resources/test_lambda.py:36]: probably a typo causing preflight linting to fail.

Hint: double-check the pre-commit hooks and/or IDE linter settings. They usually catch these, but I've also had some escaping into CI (especially around linter updates) :)

Comment (Morijarti) [reply tests/aws/services/cloudformation/resources/test_lambda.py]: Fixed

File: tests/aws/services/cloudformation/resources/test_lambda.py [hunk 2]
```diff
@@ -50,6 +50,50 @@ def _assert_single_lambda_call():
     retry(_assert_single_lambda_call, retries=30)
 
 
+# TODO make a test simular to one above but for updated filtering
+
+
+@markers.snapshot.skip_snapshot_verify(
+    ["$..EventSourceMappings..FunctionArn", "$..EventSourceMappings..LastProcessingResult"]
```
Comment (joe4dev) [inline tests/aws/services/cloudformation/resources/test_lambda.py:57]: Great catch, which will be helpful for the ESM rework 👍

File: localstack/services/lambda_/resource_providers/aws_lambda_eventsourcemapping.py [hunk 3]
```diff
@@ -180,4 +180,36 @@ def update(
           - lambda:UpdateEventSourceMapping
           - lambda:GetEventSourceMapping
         """
-        raise NotImplementedError
```
Comment (joe4dev) [inline localstack/services/lambda_/resource_providers/aws_lambda_eventsourcemapping.py:183]: Glad to see some more updates supported, especially for highly demanded functionality 📈

File: tests/aws/services/cloudformation/resources/test_lambda.py [hunk 4]
```diff
@@ -50,6 +50,50 @@ def _assert_single_lambda_call():
...
+)
+@markers.aws.validated
+def test_lambda_w_dynamodb_event_filter_update(deploy_cfn_template, snapshot, aws_client):
```
Comment (joe4dev) [inline tests/aws/services/cloudformation/resources/test_lambda.py:60]: Useful test with clear intention 👏👏👏

**Review by joe4dev — APPROVED** [review]
LGTM, well done @Morijarti 👏👏👏  🚀 

Good catch with the deprecated Lambda runtime aside 🕵️

**Pull request merged**